spec/models/solr_document_spec.rb

```diff
@@ -124,6 +124,42 @@
     end
   end
 
+  describe '#eresources_library_display_name' do
+    eresource_bus_library = { id: 30, item_display_struct:
+                [{ barcode: '9102385731231', callnumber: nil, current_location: nil,
+                   effective_location_id: '43c011f3-adeb-4c49-b357-06772aff11de',
+                   full_shelfkey: nil, home_location: 'BUS-ELECTRONIC', library: 'BUSINESS',
+                   permanent_location_code: 'BUS-ELECTRONIC', scheme: 'LC', type: 'ONLINE' }] }
+
+    context 'when it has holdings but no phsyical items' do
+      let(:solr_doc) { SolrDocument.new(eresource_bus_library) }
+
+      it 'is has a eresources_library_display_name' do
+        expect(solr_doc.eresources_library_display_name).to eq('Business Library')
+      end
+    end
+
+    context 'when it has an eresouce and physical copies' do
+      physical_location_hash = eresource_bus_library.deep_dup
+      physical_location_hash[:item_display_struct].push({ barcode: 123456, home_location: 'BUS', library: 'BUSINESS',
+                                                          permanent_location_code: 'BUS' })
+      let(:physical_location_doc) { SolrDocument.new(physical_location_hash) }
+
+      it 'has an eresource and phsyical location' do
+        expect(physical_location_doc.eresources_library_display_name).to be_nil
+      end
+    end
+
+    context 'when it is a SUL-ERESOURCE' do
```

Comment:
I'm still having trouble understanding which case this is.  
```suggestion
    context 'when there are physical items and the library is SUL' do
```